micro-rdk/src/common/analog.rs

```diff
@@ -42,6 +43,19 @@ pub trait AnalogReader<Word> {
     fn name(&self) -> String;
 }
 
+impl<A, Word> AnalogReader<Word> for Arc<Mutex<A>>
+where
+    A: ?Sized + AnalogReader<Word>,
+{
+    type Error = A::Error;
+    fn read(&mut self) -> Result<Word, Self::Error> {
+        self.lock().unwrap().read()
+    }
+    fn name(&self) -> String {
+        self.lock().unwrap().name()
+    }
+}
+
```

Comment:
Should there be a `pub AnalogReaderType`? Something like `pub type AnalogReaderType<W, E = AnalogError> = Arc<Mutex<dyn AnalogReader<W, Error = E>>>;` That'd let you simplify declarations elsewhere, if it works.

Reply:
yes! I meant to do this to match the I2CHandle pattern but I forgot